### PR TITLE
[DEVREL-1153] feat(sentimental-jeff-app) update custom field

### PR DESCRIPTION
**Pull request by ravirajsubramanian**
Update a custom ticket field with the sentiment analysis result

### CODE REVIEW

**Review by kaustavdm — CHANGES_REQUESTED** [review]
Woah. This is a complex PR. Excellent job though!

I've nitpicked some typos which you can change, and added some comments about using loops, which you can choose to act on. Code complexity is quite high in some functions. But looks like it does the job overall.

Some inline comments explaining what is happening would help anyone reading the code.

File: Demo-Apps/sentimental_jeff_app/app/app.js [hunk 1]
```diff
@@ -1,123 +1,215 @@
+const SENTIMENTS = {
+  good: 'good',
+  bad: 'bad',
+  nuetral: 'nuetral'
```
[the remark this answers is not in the document]
Comment (ravirajsubramanian) [reply Demo-Apps/sentimental_jeff_app/app/app.js]: Changed, thanks for catching it. 😃

File: Demo-Apps/sentimental_jeff_app/app/app.js [hunk 2]
```diff
@@ -1,123 +1,215 @@
...
+            status: ticket.status,
+            custom_fields: { [iparams.sentimentField]: sentiment }
+          })
+        };
+        client.request.put(url, options).then(() => {
+          console.info('Success: Updated the ticket field with the sentiment analysis result');
+        }, error => {
+          console.error('Error: Unable to update the ticket field with the sentiment analysis result');
+          console.error(error);
+          notifyError('Failed to update sentiment result to the configed ticket field.')
```
[the remark this answers is not in the document]
Comment (ravirajsubramanian) [reply Demo-Apps/sentimental_jeff_app/app/app.js]: Changed. 👍

File: Demo-Apps/sentimental_jeff_app/app/app.js [hunk 3]
```diff
@@ -1,123 +1,215 @@
...
-      sentiment = "angry";
+  while (emoticonsLen--) {
+    let emoticon = emoticonTokens[emoticonsLen];
+    if (!AFINN_EMOTICON.hasOwnProperty(emoticon)) {
+      continue;
     }
-    sentiment = 'emoji--' + sentiment;
-    return sentiment;
+    score += AFINN_EMOTICON[obj];
   }
```
[the remark this answers is not in the document]
Comment (ravirajsubramanian) [reply Demo-Apps/sentimental_jeff_app/app/app.js]: Changed it the same way. The only change is `const` within the for loop will break in Firefox it seems. So, initializing the variable before the loop.

File: Demo-Apps/sentimental_jeff_app/README.md [hunk 4]
```diff
@@ -2,31 +2,32 @@
...
 ## Features Demonstrated
 
-App location: ticket_sidebar on ticket details page
+Product: Freshdesk
+App location: ticket_sidebar on Ticket Details Page
 
 | Feature | Notes |
 | :---: | --- |
-| [`Data API`](https://developers.freshdesk.com/v2/docs/data-api/) | Data API is used to retrieve _ticket id_ and _Domain Name_ |
-| [`Request API`](https://developers.freshdesk.com/v2/docs/request-api/) | Request API has been used to get the conversational details |
```
Comment (Saif-Shines) [inline Demo-Apps/sentimental_jeff_app/README.md:24]: I appreciate the not forgetting to update README following the enhancements to the app :)

**Review by Saif-Shines — APPROVED** [review]
:)

File: Demo-Apps/sentimental_jeff_app/app/app.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaustavdm) [reply Demo-Apps/sentimental_jeff_app/app/app.js]: That's weird. You should still be able to use the `let` inside the `for` loop.

On looking up, I found [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1237395) filed, but looks like it was resolved years ago. I also tried `for...of` with `const` in Firefox console, and it worked.

Screenshot 2021-05-18 at 1 01 54 PM

Can you see if you can reproduce this error on Firefox? If yes, see if you can bump up that thread in Bugzilla.

**Review by kaustavdm — DISMISSED** [review]
Approving for now. But take a look at the `const` usage in `for..of`.

File: Demo-Apps/sentimental_jeff_app/app/app.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ravirajsubramanian) [reply Demo-Apps/sentimental_jeff_app/app/app.js]: Just found until Firefox version 51, the issue existed and now fixed. Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#browser_compatibility

So, just avoided using it for a wider spectrum of support browser versions.

**Review by Saif-Shines — APPROVED** [review]

**Pull request merged**